### ISSUE TITLE
Non-determinism while using Conv1D and AdaptiveMaxPool1D

### ISSUE THREAD

**Issue by Genius1237** [pytorch/pytorch#24011]
## 🐛 Bug

Non-determinism while using the aforementioned layers. Check the code sample below to look at the model.

## To Reproduce

```
class CNN(nn.Module):

    def __init__(self, input_vocab_size, embedding_hidden_dims, n_filters, n_classes):
        super(CNN, self).__init__()

        self.input_vocab_size = input_vocab_size
        self.n_filters = n_filters
        self.embedding_hidden_dims = embedding_hidden_dims
        self.n_classes = n_classes

        self.input_embedding = nn.Embedding(input_vocab_size + 1, embedding_hidden_dims, padding_idx=0)

        self.c2 = nn.Conv1d(embedding_hidden_dims, 3 * n_filters, 2)
        self.p2 = nn.AdaptiveMaxPool1d(1)
        self.c3 = nn.Conv1d(embedding_hidden_dims, 2 * n_filters, 3)
        self.p3 = nn.AdaptiveMaxPool1d(1)
        self.c4 = nn.Conv1d(embedding_hidden_dims, n_filters, 4)
        self.p4 = nn.AdaptiveMaxPool1d(1)
	
	self.linear_1 = nn.Linear(n_filters + (2 * n_filters) + (3 * n_filters), 500)
        self.linear_2 = nn.Linear(500, n_classes)

    def forward(self, source):
	source_padded = nn.utils.rnn.pad_sequence(source, batch_first=True)
	source_input = self.input_embedding(source_padded)
	c = source_input.transpose(1, 2)
	
	c2 = self.c2(c)
        c2 = self.p2(c2)
	c3 = self.c3(c)
	c3 = self.p3(c3)
	c4 = self.c4(c)
	c4 = self.p4(c4)

        c2 = c2.reshape(c2.shape[0], c2.shape[1])
        c3 = c3.reshape(c3.shape[0], c3.shape[1])
        c4 = c4.reshape(c4.shape[0], c4.shape[1])

        t = torch.cat([c2, c3, c4], dim=1)
        t = F.relu(t)
        t = F.dropout(t)
        t = self.linear_1(t)
        t = F.relu(t)
        output = self.linear_2(t)
        return output
```

For this code sample, train and save model to disk. Reload the model, set `model.eval()` and run the same example 2 times through it and you'll get 2 different outputs.

Is there anything in particular about the layers used here? I know that there is some dropout used, but `model.eval()` should take care of it. 

Also, I made the model run on CPU only and got the same bug. 

Also, I set `torch.backends.cudnn.benchmark = True` and got the same bug

## Expected behavior

Already mentioned above.

## Environment

 - PyTorch Version (e.g., 1.0): 1.1.0
 - OS (e.g., Linux): Ubuntu 16.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.6.8
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: Tesla P100
 - Any other relevant information:

**Comment by rmcavoy** [comment]
This is not a bug. You never fixed any of the random seeds. In deep learning the weights of neural networks are initialized by drawing from random number generators so you should not expect determinism if you don't fix the random seeds.

**Comment by Genius1237** [comment]
@rmcavoy I feel that you haven't understood my bug properly. What you are describing is determinism within the training process. My problem is somewhat different.

**Comment by ezyang** [comment]
Could you post a more complete code sample, including how you ran the model the first time, and then how you saved and loaded it and ran the second time?

**Comment by Genius1237** [comment]
@ezyang, I had made a mistake and that was the cause for the bug. The issue can be closed.
